gremlin-groovy/src/main/java/org/apache/tinkerpop/gremlin/groovy/engine/GremlinExecutor.java

```diff
@@ -258,7 +260,9 @@ public CompletableFuture<Object> eval(final String script, final String language
 
         // override the timeout if the lifecycle has a value assigned. if the script contains with(timeout)
         // options then allow that value to override what's provided on the lifecycle
-        final Optional<Long> timeoutDefinedInScript = GremlinScriptChecker.parse(script).getTimeout();
+        final Optional<Long> timeoutDefinedInScript = script instanceof String
+                ? GremlinScriptChecker.parse((String) script).getTimeout()
+                : Optional.empty();
```

Comment:
Would we want to support `g.with("evaluationTimeout", 123)` for over bytecode as well as scripts?

Reply:
good catch